src/main/kotlin/com/hj/leetcode/kotlin/problem2090/Solution.kt

```diff
@@ -0,0 +1,38 @@
+package com.hj.leetcode.kotlin.problem2090
+
+/**
+ * LeetCode page: [2090. K Radius Subarray Averages](https://leetcode.com/problems/k-radius-subarray-averages/);
+ */
+class Solution {
+    /* Complexity:
+     * Time O(N) and Space O(N) where N is the size of nums;
+     */
+    fun getAverages(nums: IntArray, k: Int): IntArray {
+        if (k == 0) {
+            return nums.clone()
+        }
+
+        val lessKRadiusAverage = -1
+        val result = IntArray(nums.size) { lessKRadiusAverage }
+
+        val validCenterIndices = k until (nums.size - k)
+        if (validCenterIndices.isEmpty()) {
+            return result
+        }
+
+        // Apply sliding window technique
+        val windowSize = k * 2 + 1
+        var windowSum = nums.sum(0 until windowSize)
+        result[k] = (windowSum / windowSize).toInt()
```

Comment:
extract function that performs the integer division

Reply:
Addressed at https://github.com/hj-core/hj-leetcode-kotlin/pull/91/commits/118a5474461be20365a57bdfcb8822be39a409ed